master/internal/scheduler/fitting.go

```diff
@@ -68,18 +70,17 @@ func (c candidateList) Swap(i, j int) {
 func findFits(
 	task *Task, agents map[*actor.Ref]*agentState, fittingMethod SoftConstraint,
 ) []*fittingState {
-	if task.SlotsNeeded() <= 1 {
-		if fit := findNotMultiSlotFit(task, agents, fittingMethod); fit != nil {
-			return []*fittingState{
-				fit,
-			}
-		}
-	} else {
-		if fits := findMultiSlotFits(task, agents, fittingMethod); len(fits) != 0 {
-			return fits
+	if !task.fittingRequirements.DedicatedAgent {
+		if fit := findSharedAgentFit(task, agents, fittingMethod); fit != nil {
```

Comment:
question: does non-dedicated d-train jobs go down this path?

Reply:
Yeah if host-mode is not the default networking all trials will first try going down this path.